spec/support/shared_examples.rb

```diff
@@ -5,3 +5,11 @@
     expect(response).to redirect_to sign_in_path
   end
 end
+
+shared_examples "requires admin" do
+  it "redirects to home path" do
+    session[:user_id] = Fabricate(:user)
```

Comment:
In session, you want to store `:user_id`, and currently you are passing in user object. It makes your session bigger, i would recommend passing in just that id.